### DISCUSSION TITLE
How to use event-based datasets(n-mnist, dvs128) without encoding?

### DISCUSSION THREAD

**Discussion by mountains-high** [fangwei123456/SpikingFlow#74] [Q&A] [answered]
Good day, Dear Developers,

Could you show the way of how to use the 'n_mnist' dataset, please?
How to encode and normalize?
Something like this:
```from spikingjelly.datasets import n_mnist ```
```train_set = n_mnist.NMNIST('.', train=True, data_type='frame', frames_number=16, split_by='number')```
When we use this, does it mean that our data already was encoded? 

Thank you!

**Comment by fangwei123456** [comment, marked as answer]
Yes, if you set data_type='frame', you will get frames, which are integrated from events. If you want to use the origin events, you can set data_type='event'.

See https://spikingjelly.readthedocs.io/zh_CN/latest/clock_driven_en/13_neuromorphic_datasets.html for more details.

**Reply by mountains-high** [reply]
Thank you very much, I will try.
